File: app/src/main/java/com/arbaelbarca/listmovieskatalog/presenter/detailmoviespresenter/DetailMoviesPresenter.java
package com.arbaelbarca.listmovieskatalog.presenter.detailmoviespresenter;

import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.GenresItem;
import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.ResponseDetailMovies;
import com.arbaelbarca.listmovieskatalog.model.modelreviewmovies.ResultsItemReviews;
import com.arbaelbarca.listmovieskatalog.view.DetailMoviesView;

import java.util.ArrayList;

public class DetailMoviesPresenter implements DetailMoviesView.presenter, DetailMoviesView.GetNoticeIntractorDetailMovies.OnFinishedListener {

    DetailMoviesView.presenter presenter;
    DetailMoviesView.MainView mainView;
    DetailMoviesView.GetNoticeIntractorDetailMovies getNoticeIntractorDetailMovies;


    public DetailMoviesPresenter(DetailMoviesView.MainView mainView, DetailMoviesView.GetNoticeIntractorDetailMovies getNoticeIntractorDetailMovies) {
        this.mainView = mainView;
        this.getNoticeIntractorDetailMovies = getNoticeIntractorDetailMovies;
    }

    @Override
    public void onDestroy() {
        mainView = null;
    }

    @Override
    public void requestFromDataServer(String movieId) {
        getNoticeIntractorDetailMovies.getDetailReviews(this, movieId);
        getNoticeIntractorDetailMovies.getDetailMovies(this, movieId);
    }

    @Override
    public void refreshData(String type) {

    }

    @Override
    public void onFinished(ResponseDetailMovies responseDetailMovies) {
        if (mainView != null)
            mainView.setDataTags(responseDetailMovies);
    }

    @Override
    public void onFailure(Throwable t) {
        if (mainView != null) {
            mainView.onResponseFailure(t);
            mainView.hideProgress();
        }
    }

    @Override
    public void setDataTags(ArrayList<GenresItem> genresItems) {

    }

    @Override
    public void setDataReview(ArrayList<ResultsItemReviews> resultsItemReviews) {
        if (mainView != null) {
            mainView.setDataReview(resultsItemReviews);
            mainView.hideProgress();
        }
    }
}

File: app/src/main/java/com/arbaelbarca/listmovieskatalog/presenter/detailmoviespresenter/DetailMoviesPresenter.kt
package com.arbaelbarca.listmovieskatalog.presenter.detailmoviespresenter

import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.GenresItem
import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.ResponseDetailMovies
import com.arbaelbarca.listmovieskatalog.model.modelreviewmovies.ResultsItemReviews
import com.arbaelbarca.listmovieskatalog.view.DetailMoviesView
import java.util.*

class DetailMoviesPresenter(internal var mainView: DetailMoviesView.MainView?, internal var getNoticeIntractorDetailMovies: DetailMoviesView.GetNoticeIntractorDetailMovies) : DetailMoviesView.presenter, DetailMoviesView.GetNoticeIntractorDetailMovies.OnFinishedListener {

    internal var presenter: DetailMoviesView.presenter? = null

    override fun onDestroy() {
        mainView = null
    }

    override fun requestFromDataServer(movieId: String) {
        getNoticeIntractorDetailMovies.getDetailReviews(this, movieId)
        getNoticeIntractorDetailMovies.getDetailMovies(this, movieId)
    }

    override fun refreshData(type: String) {

    }

    override fun onFinished(responseDetailMovies: ResponseDetailMovies) {
        if (mainView != null)
            mainView!!.setDataTags(responseDetailMovies)
    }

    override fun onFailure(t: Throwable) {
        if (mainView != null) {
            mainView!!.onResponseFailure(t)
            mainView!!.hideProgress()
        }
    }

    override fun setDataTags(genresItems: ArrayList<GenresItem>) {

    }

    override fun setDataReview(resultsItemReviews: ArrayList<ResultsItemReviews>) {
        if (mainView != null) {
            mainView!!.setDataReview(resultsItemReviews)
            mainView!!.hideProgress()
        }
    }
}
